pallets/oracle/src/tests.rs

```diff
@@ -0,0 +1,349 @@
+	RuntimeAppPublic,
+};
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// For testing the module, we construct a mock runtime.
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		Oracle: pallet_oracle::{Pallet, Call, Event<T>, ValidateUnsigned},
+	}
+);
+
+parameter_types! {
+	pub BlockWeights: frame_system::limits::BlockWeights =
+		frame_system::limits::BlockWeights::simple_max(frame_support::weights::Weight::from_ref_time(1024));
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
+	pub StorageBytesMultiplier: u64 = 10;
+	pub const IsTransferable: bool = false;
+}
+impl frame_system::Config for Test {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type RuntimeOrigin = RuntimeOrigin;
+	type RuntimeCall = RuntimeCall;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = sp_core::ed25519::Public;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type RuntimeEvent = RuntimeEvent;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = SS58Prefix;
+	type OnSetCode = ();
+	type MaxConsumers = ConstU32<2>;
+}
+
+type Extrinsic = TestXt<RuntimeCall, ()>;
+type AccountId = <<Signature as Verify>::Signer as IdentifyAccount>::AccountId;
+
+impl frame_system::offchain::SigningTypes for Test {
+	type Public = <Signature as Verify>::Signer;
+	type Signature = Signature;
+}
+
+impl<LocalCall> frame_system::offchain::SendTransactionTypes<LocalCall> for Test
+where
+	RuntimeCall: From<LocalCall>,
+{
+	type Extrinsic = Extrinsic;
+	type OverarchingCall = RuntimeCall;
+}
+
+impl<LocalCall> frame_system::offchain::CreateSignedTransaction<LocalCall> for Test
+where
+	RuntimeCall: From<LocalCall>,
+{
+	fn create_transaction<C: frame_system::offchain::AppCrypto<Self::Public, Self::Signature>>(
+		call: RuntimeCall,
+		_public: <Signature as Verify>::Signer,
+		_account: AccountId,
+		nonce: u64,
+	) -> Option<(RuntimeCall, <Extrinsic as ExtrinsicT>::SignaturePayload)> {
+		Some((call, (nonce, ())))
+	}
+}
+
+impl Config for Test {
+	type AuthorityId = crypto::FragAuthId;
+	type RuntimeEvent = RuntimeEvent;
+	type OracleContract = Test;
+	type Threshold = ConstU64<1>;
+}
+
+impl OracleContract for Test {
+	fn get_contract() -> &'static str {
+		// https://docs.chain.link/docs/data-feeds/price-feeds/addresses/
+		"0x547a514d5e3769680Ce22B2361c10Ea13619e8a9" // AAVE / USD on mainnet (this is just for testing purposes)
```

Comment:
I don't think we will ultimately use chainlink to be honest. Do you think we could convert this oracle to use uniswap instead?

Reply:
But in general we need to rethink a bit, let's leave this here as is.. but would be probably good to have multiple contracts and multiple abis. So maybe that's the needed extra bit.

Reply:
I can make this to be `Vec` of contracts addresses and the logic could use the first one as default (i.e. Uniswap), falling back to the second contract (i.e. Chainlink) in the list in case the first one is not responding (i.e. timeout). Would that work?

Reply:
it will need to be an enum specifying which abi to use